map/src/drawer/components/tracks/CloudTrackItem.jsx

```diff
@@ -0,0 +1,78 @@
+import AppContext from "../../../context/AppContext";
+import {ListItemText, MenuItem, Switch, Tooltip, Typography} from "@mui/material";
+import React, {useContext} from "react";
+import Utils from "../../../util/Utils";
+import TrackInfo from "./TrackInfo";
+import TrackUtils from "../../util/TrackUtils";
+
+export default function CloudTrackItem({file}) {
+
+    const ctx = useContext(AppContext);
+
+    async function enableLayer(setProgressVisible, visible) {
+        if (!visible) {
+            deleteTrackFromMap();
+        } else {
+            await addTrackToMap(setProgressVisible);
+        }
+    }
+
+    function deleteTrackFromMap() {
+        const newGpxFiles = Object.assign({}, ctx.gpxFiles);
+        newGpxFiles[file.name].url = null;
+        ctx.setGpxFiles(newGpxFiles);
+        if (ctx.selectedGpxFile?.name === file.name) {
+            ctx.setContextMenuObjectType(null);
+        }
+        TrackUtils.updateTextInfo(newGpxFiles, ctx);
```

Comment:
updateTextInfo - showed selected tracks. 1) code already doesn't work for locals 2) concurrent actions. 
Text info should be updated once by Menu part i.e. inside TracksMenu

Reply:
`+`